appserver/batch/glassfish-batch-connector/src/main/java/fish/payara/jbatch/persistence/rdbms/JBatchJDBCPersistenceManager.java

```diff
@@ -213,7 +213,7 @@ public CheckpointData getCheckpointData(CheckpointDataKey key) {
 
         tryObtainTableLock();
 
-        CheckpointData checkpointData = queryCheckpointData(key.getCommaSeparatedKey());
+        CheckpointData checkpointData = key == null ? null : queryCheckpointData(key.getCommaSeparatedKey());
```

Comment:
While the logging in the method suggest that `key` can be null I believe it is not really expected that callers use null keys. The query performed is a "select where id =" which does not work for null so I concluded that we simply don't even try and finish with `null` data for `null` key which is logged and handled fine by the rest of the method.